src/fixate/_switching.py

```diff
@@ -87,14 +92,26 @@ def __or__(self, other: PinUpdate) -> PinUpdate:
 
 PinUpdateCallback = Callable[[PinUpdate, bool], None]
 
+S = TypeVar("S", bound=str)
 
-class VirtualMux:
+class VirtualMux(Generic[S]):
     pin_list: PinList = ()
     clearing_time: float = 0.0
 
+    def __class_getitem__(cls, *args, **kwargs):
+        # without calling getitem the class doesn't work as a generic
+        getitm =  super().__class_getitem__(*args, **kwargs) # normally returns a generic
+        # create a proxy to force the __init_subclass__ hook
+        class Hack(getitm):
```

Comment:
Perhaps rename the `Hack` to `SpecialisedVirtualMux`? Something that reflects its role.

Also need to use `functools.wraps` or whatever secret sauce is required to make errors nice

Reply:
We can revisit this idea when we have a better way of getting around GenericAlias stripping all annotations at runtime.